pkg/didcomm/protocol/didexchange/service_test.go

```diff
@@ -903,80 +939,77 @@ func TestService_Execute(t *testing.T) {
+	ch := make(chan dispatcher.DIDCommEvent)
+	err := svc.RegisterActionEvent(ch)
 	require.NoError(t, err)
 
+	// register another action event
+	err = svc.RegisterActionEvent(make(chan dispatcher.DIDCommEvent))
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "channel is already registered for the action event")
+
 	// validate after register
 	require.NotNil(t, svc.actionEvent)
-	require.True(t, svc.execute)
 
 	// unregister a action event
-	err = svc.UnregisterEvent()
+	err = svc.UnregisterActionEvent(ch)
 	require.NoError(t, err)
 
 	// validate after unregister
 	require.Nil(t, svc.actionEvent)
-	require.False(t, svc.execute)
-
-	// register for auto execute
-	err = svc.RegisterAutoExecute()
-	require.NoError(t, err)
-	require.True(t, svc.execute)
-
-	// unregister for auto execute
-	err = svc.UnregisterAutoExecute()
-	require.NoError(t, err)
-	require.False(t, svc.execute)
-
-	// register for auto execute and action event
-	err = svc.RegisterEvent(ch)
-	require.NoError(t, err)
-	err = svc.RegisterAutoExecute()
-	require.NoError(t, err)
-	require.True(t, svc.execute)
 
-	// unregister action event with auto execute set
-	err = svc.UnregisterEvent()
-	require.NoError(t, err)
-	require.False(t, svc.execute)
+	// unregister with different channel
+	err = svc.UnregisterActionEvent(make(chan dispatcher.DIDCommEvent))
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "invalid channel passed to unregister the action event")
 }
 
-func TestService_StatusEvents(t *testing.T) {
-
+func TestService_MsgEvents(t *testing.T) {
 	dbstore, cleanup := store(t)
 	defer cleanup()
 
 	svc := New(dbstore, &mockProvider{})
 
 	// validate before register
-	require.Nil(t, svc.statusEvents)
-	require.Equal(t, 0, len(svc.statusEvents))
+	require.Nil(t, svc.msgEvents)
+	require.Equal(t, 0, len(svc.msgEvents))
 
 	// register a status event
-	ch := make(chan dispatcher.DIDCommMsg, 10)
-	err := svc.RegisterMsg(ch)
+	ch := make(chan dispatcher.DIDCommMsg)
+	err := svc.RegisterMsgEvent(ch)
 	require.NoError(t, err)
 
 	// validate after register
-	require.NotNil(t, svc.statusEvents)
-	require.Equal(t, 1, len(svc.statusEvents))
+	require.NotNil(t, svc.msgEvents)
+	require.Equal(t, 1, len(svc.msgEvents))
 
 	// register a new status event
-	err = svc.RegisterMsg(make(chan dispatcher.DIDCommMsg, 10))
+	err = svc.RegisterMsgEvent(make(chan dispatcher.DIDCommMsg))
 	require.NoError(t, err)
 
 	// validate after new register
-	require.NotNil(t, svc.statusEvents)
-	require.Equal(t, 2, len(svc.statusEvents))
+	require.NotNil(t, svc.msgEvents)
+	require.Equal(t, 2, len(svc.msgEvents))
 
 	// unregister a status event
-	err = svc.UnregisterMsg(ch)
+	err = svc.UnregisterMsgEvent(ch)
 	require.NoError(t, err)
 
 	// validate after unregister
-	require.Equal(t, 1, len(svc.statusEvents))
+	require.Equal(t, 1, len(svc.msgEvents))
+}
+
+func Test_AutoExecute(t *testing.T) {
+	ch := make(chan dispatcher.DIDCommEvent)
+	done := make(chan bool)
```

Comment:
FYI: done channel signatures are usually this:
`chan struct{}`
the bool is not needed.

You signal done with `close(chName)`.
You wait for done with `<-done`

Reply:
[Updated](https://github.com/hyperledger/aries-framework-go/compare/eeed66994693b7c17d4b1c90ad61e09f2e22cbe7..4d12dff2d14ec87906c8c69146befde29694ebfe#diff-36ac39bebf32030fd7dff6f12aea8d7eR1006).